File: src/main/java/com/empresa/logistica/service/PedidoService.java
package com.empresa.logistica.service;

import com.empresa.logistica.dto.PedidoDTO;
import com.empresa.logistica.dto.request.CriarPedidoRequest;
import com.empresa.logistica.model.StatusPedido;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * Service interface for Pedido business logic
 */
public interface PedidoService {
    
    /**
     * Creates a new order with credit limit validation
     */
    PedidoDTO criarPedido(CriarPedidoRequest request);
    
    /**
     * Find order by ID
     */
    PedidoDTO buscarPorId(Long id);
    
    /**
     * List all orders with pagination
     */
    Page<PedidoDTO> listarPedidos(Pageable pageable);
    
    /**
     * List orders by customer with pagination
     */
    Page<PedidoDTO> listarPorCliente(Long clienteId, Pageable pageable);
    
    /**
     * List orders by status with pagination
     */
    Page<PedidoDTO> listarPorStatus(StatusPedido status, Pageable pageable);
    
    /**
     * Calculate total orders for customer in date range
     */
    BigDecimal calcularTotalPorCliente(Long clienteId, LocalDateTime dataInicio, LocalDateTime dataFim);
}

File: frontend/src/services/pedidoService.js
import api from './api';

// Pedido service for managing order data
export const pedidoService = {
  // Get all orders with pagination
  getAll: async (page = 0, size = 20) => {
    const response = await api.get(`/pedidos?page=${page}&size=${size}`);
    return response.data;
  },

  // Get order by ID
  getById: async (id) => {
    const response = await api.get(`/pedidos/${id}`);
    return response.data;
  },

  // Get orders by client ID
  getByCliente: async (clienteId, page = 0, size = 20) => {
    const response = await api.get(`/pedidos/cliente/${clienteId}?page=${page}&size=${size}`);
    return response.data;
  },

  // Get orders by status
  getByStatus: async (status, page = 0, size = 20) => {
    const response = await api.get(`/pedidos/status/${status}?page=${page}&size=${size}`);
    return response.data;
  },

  // Create new order (main business logic function)
  create: async (pedidoData) => {
    const response = await api.post('/pedidos', pedidoData);
    return response.data;
  },

  // Get client's total orders in last 30 days
  getClienteTotal30Dias: async (clienteId) => {
    const response = await api.get(`/pedidos/cliente/${clienteId}/total`);
    return response.data;
  },

  // Filter orders by date range
  filterByDate: async (dataInicio, dataFim, page = 0, size = 20) => {
    const response = await api.get(`/pedidos/periodo?inicio=${dataInicio}&fim=${dataFim}&page=${page}&size=${size}`);
    return response.data;
  }
};
